src/structures/MessageEmbed.js

```diff
@@ -341,7 +344,7 @@ class MessageEmbed {
    * @returns {MessageEmbed}
    */
   setImage(url) {
-    this.image = { url };
+    this.image = { url: String(url) };
```

Comment:
Why are these not verified?

Reply:
If I remember my reasoning correctly these  (URLs) are rejected by the API if anything but a valid looking URL is provided, with an appropriate error, leading to no surprising effects (like the mentioned "undefined", "null" and "[Object object]" strings in places where people pass non-string structures.

That being said I could see how verifying the string on our end and then still having the verification of URL format on the API could work in a layered sense.